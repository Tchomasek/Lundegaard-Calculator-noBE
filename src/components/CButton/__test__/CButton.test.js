import { CButton } from "../index";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";

import { cleanup, render } from "@testing-library/react";

afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<CButton />, div);
});

it("renders button correctly", () => {
  const { getByTestId } = render(<CButton info={"Click"} />);
  expect(getByTestId("CButton")).toHaveTextContent("Click");
});

it("renders button correctly", () => {
  const { getByTestId } = render(<CButton info={"Save"} />);
  expect(getByTestId("CButton")).toHaveTextContent("Save");
});

it("matches snapshot", () => {
  const button = renderer.create(<CButton info={"Click"} />).toJSON();
  expect(button).toMatchSnapshot();
});
