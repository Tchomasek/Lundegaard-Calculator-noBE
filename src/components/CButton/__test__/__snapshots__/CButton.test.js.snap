// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
  data-testid="CButton"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    Click
  </span>
</button>
`;
